### DISCUSSION TITLE
How do I submit the PR to GitHub Octernships?

### DISCUSSION THREAD

**Discussion by ashwinexe** [GitHub-Octernships/Ivy-Octernships-ML#4] [Q&A] [answered]
This discussion will answer all your queries regarding submission to Ivy Octernships

Feel free to drop more questions regarding submissions below.

**Comment by ashwinexe** [comment, marked as answer]
## How the Pull Request submission works?
There may be some confusion regarding submission at Ivy Octernships. 
If you head over to [Task Submission](https://github.com/GitHub-Octernships/Ivy-Octernships-ML#task-submission-) of this Octernships you may see we ask you to create and merge PR for submission. 

**You will need to create and merge PR (on or before deadline) in your private submission repository for a valid submission**
Your private repository will follow this naming pattern: `ivy-octernships-ml-USERNAME`

_Only commits on this repository will be a considered for review for GitHub Octernships._

## Why do we not allow direct PRs to the project?
GitHub Octernships is launched to provide paid professional mentorship to **verified students** who are also actively contributing to open source on GitHub. Since anyone can create a public PR, we cannot validate public PRs as submissions to Octernships.

## Should I use the template repository to generate a template for myself and start working?
Generating template from the assignment repository is similar to forking a project. **We do not allow Octernships submissions done via forking or personally generated templates.** A private submission repository is automatically created for you when you apply to GitHub Octernships from the [Octernships Dashboard](https://education.github.com/globalcampus/octernships)

## How do I know if I am submitting in the right place?
Your will have access to the repository: `ivy-octernships-ml-USERNAME` in this Organization. Please submit your assignment in that repository only. We have setup the repo with all boiler-plate code to help you complete the task.

You are free to contribute to open source projects but public contribution cannot be used as assignment submissions.

Feel free to drop more questions regarding submissions below.

**Reply by ManoBharathi93** [reply]
After applying via [Octernships Dashboard](https://education.github.com/globalcampus/octernships)
I get this repository
<img width="843" alt="image" src="https://github.com/GitHub-Octernships/Ivy-Octernships-ML/assets/88357044/ab76b44f-8f43-4a3b-b374-ad949c166023">
Step: 1  Click on Use on template  and make a repository (whether I can create public or private repo) ? - name os repo ivy-octernships-ml-ManoBharathi93 (in my case)
Step  2: Clone  my repo that i created in step -1 
step 3:  In local create a new branch and make changes.
Step 4:  Make a PR to my repo and merge it.. 



Am I right @ashwinexe

**Reply by ashwinexe** [reply]
You cannot create a private repository. GitHub does it for you when you apply officially via [Octernships](https://education.github.com/students/octernships)

**Reply by ManoBharathi93** [reply]
<img width="953" alt="image" src="https://github.com/GitHub-Octernships/Ivy-Octernships-ML/assets/88357044/26c80c81-43a6-4185-8a11-58e1330a0a42">
Yes I got it.. I will clone this repo and make new branch and complete the task and make a PR

**Reply by bilal-aamer** [reply]
We are supposed to merge the new-branch with the master right as part of the task submission. Can we do multiple private PR merges for multiple tasks?

**Comment by Anubhav-Goyal01** [comment]
Hi @ashwinexe, I applied via the GitHub Octernship page and it did create a private repo for me, I created and merged a PR in that private repo of mine, is that enough for making it a valid submission or do I need to do something else?

**Reply by poemsforaphrodite** [reply]
Did you pass all of the tests?

**Reply by Anubhav-Goyal01** [reply]
No, only 4 got passed, still working on it.

**Comment by Gauravpadam** [comment]
Hey @ashwinexe, I'm facing an issue while trying to merge my PR
It closes automatically and displays this message
![image](https://github.com/GitHub-Octernships/Ivy-Octernships-ML/assets/12787482/88e8eec8-1abc-4f48-9f09-35cfa089900d)

Would you please help? Thanks!

**Reply by p1utoze** [reply]
I second this, I created a PR passing all tests but it closed automatically with the same message.

**Reply by ashwinexe** [reply]
Uh oh! you may wanna delete the `auto-close-pr.yaml` in `.github/workflows` of your repository. Let me know if that works.
